portal/src/components/roles-and-groups/list/UserDetailsScreenGroupListContainer.tsx

```diff
@@ -0,0 +1,162 @@
+  const { renderToString } = useContext(MessageContext);
+  const {
+    data: groupsQueryData,
+    loading,
+    error,
+    refetch,
+  } = useQuery<GroupsListQueryQuery, GroupsListQueryQueryVariables>(
+    GroupsListQueryDocument,
+    {
+      variables: {
+        pageSize: 0,
+        searchKeyword: "",
+      },
+      fetchPolicy: "network-only",
+    }
+  );
+
+  const [searchKeyword, setSearchKeyword] = useState<string>("");
+  const isSearch = searchKeyword !== "";
+  const [offset, setOffset] = useState(0);
+
+  const onChangeOffset = useCallback((offset) => {
+    setOffset(offset);
+  }, []);
+
+  const onChangeSearchKeyword = useCallback(
+    (e?: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      if (e === undefined) {
+        return;
+      }
+      const value = e.currentTarget.value;
+      setSearchKeyword(value);
+      // Reset offset when search keyword was changed.
+      setOffset(0);
+    },
+    []
+  );
+  const onClearSearchKeyword = useCallback(() => {
+    setSearchKeyword("");
+  }, []);
+
+  const [isAddRoleDialogHidden, setIsAddRoleDialogHidden] = useState(true);
+  const showAddRoleDialog = useCallback(
+    () => setIsAddRoleDialogHidden(false),
+    []
+  );
+  const hideAddRoleDialog = useCallback(
+    () => setIsAddRoleDialogHidden(true),
+    []
+  );
+
+  const filteredUserGroups = useMemo(() => {
+    const userGroups =
+      user.groups?.edges?.flatMap<UserGroupsListItem>((edge) => {
+        if (edge?.node != null) {
+          return [edge.node];
+        }
+        return [];
+      }) ?? [];
+    if (isSearch) {
+      return searchGroups(userGroups, searchKeyword);
+    }
+
+    return userGroups.slice(offset, offset + pageSize);
+  }, [user.groups?.edges, isSearch, offset, searchKeyword]);
+
+  const userGroups = useMemo(() => {
+    return (
+      user.groups?.edges?.flatMap((e) => {
+        if (e?.node) {
+          return [e.node];
+        }
+        return [];
+      }) ?? []
+    );
+  }, [user.groups?.edges]);
+
+  if (error != null) {
+    return <ShowError error={error} onRetry={refetch} />;
+  }
+
+  if (loading) {
+    return <ShowLoading />;
+  }
+
+  const totalCount = groupsQueryData?.groups?.totalCount ?? 0;
+
+  if (totalCount === 0) {
+    return <GroupsEmptyView />;
+  }
+
+  return (
+    <>
+      <section className={cn("flex flex-col h-full", className)}>
+        <header className="flex flex-row items-center justify-between mb-8">
+          <SearchBox
+            className="max-w-[300px] min-w-0 flex-1 mr-2"
+            placeholder={renderToString("search")}
+            value={searchKeyword}
+            onChange={onChangeSearchKeyword}
+            onClear={onClearSearchKeyword}
+          />
+          <PrimaryButton
+            text={<FormattedMessage id="UserDetailsScreen.groups.add" />}
+            onClick={showAddRoleDialog}
+          />
+        </header>
+        <UserGroupsList
+          className="flex-1 min-h-0"
```

Comment:
<img width="1051" alt="Screenshot 2024-03-18 at 7 40 49 PM" src="https://github.com/authgear/authgear-server/assets/34942210/766c283d-2fff-468f-95d5-267553faa869">

When the screen is too short, the list could become completely hidden.
So lets give it a minimum height, say 200px.
Also, use `flex-1-0-auto` to ensure it doesn't shrink beyond the minimum height.